decimal.go

```diff
@@ -1102,6 +1104,8 @@ func (z *Big) SetString(s string) (*Big, bool) {
 				return nil, false
 			}
 			err = nerr.Err
+		} else if z.compact == 0 {
```

Comment:
Unless we can come up with a test case where this matters one way or the other, I'm comfortable leaving it as is.